src/server/dfly_main.cc

```diff
@@ -609,6 +610,16 @@ void sigill_hdlr(int signo) {
   exit(1);
 }
 
+void PrintBasicUsageInfo() {
+  const string flags_log_dir = FLAGS_log_dir.CurrentValue();
+  const string log_dir = (flags_log_dir == string("")) ? string("/tmp/") : flags_log_dir;
```

Comment:
The thing is, different systems (like MacOS? BSD variants? Windows? I don't even know :) might use different paths, so I find it preferable to use the built-in `GetLoggingDirectories()`. Even if you take the first entry there, it'll probably be more accurate than `/tmp`

Reply:
So what you write here is a :gem: because it shows one of the underline issues of defaulting `log_dir`. So my thoughts are:

1. For windows (no docker) I don't think we should care because I doubt if we can actually compile + run dragonfly even with WSL2 (because as far as I know it does not support `IO_URING` and helio won't work).

Now let's look a little bit on the nuanced details: 
 
 ```
    const vector<string>& GetLoggingDirectories() {
    // Not strictly thread-safe but we're called early in InitGoogle().
    if (logging_directories_list == NULL) {
      logging_directories_list = new vector<string>;
  
      if ( !FLAG(log_dir).empty() ) {                                                 // ** 1 **
        // A dir was specified, we should use it
        logging_directories_list->push_back(FLAG(log_dir));
      } else {                                                                        // ** 2 **
        GetTempDirectories(logging_directories_list);
  #ifdef GLOG_OS_WINDOWS
        char tmp[MAX_PATH];
        if (GetWindowsDirectoryA(tmp, MAX_PATH))
          logging_directories_list->push_back(tmp);
        logging_directories_list->push_back(".\\");
  #else
        logging_directories_list->push_back("./");                                    // ** 3 **
  #endif
      }    
    }
    return *logging_directories_list;
  }
  ```
  
  * For (1) is true we just output the directory specified on `--logs_dir` --> All good here
  * For (2) we call `GetTempDirectories()` and then we either push windows directories or the `"./"`  for the rest of the oses. 
  
  Now let's take a look at `GetTempDirectories()`  (I strip away the windows specific code)
  
  ```
      // Directories, in order of preference. If we find a dir that
    // exists, we stop adding other less-preferred dirs
    const char * candidates[] = {
      // Non-null only during unittest/regtest
      getenv("TEST_TMPDIR"),
  
      // Explicitly-supplied temp dirs
      getenv("TMPDIR"), getenv("TMP"),
  
      // If all else fails
      "/tmp",
    };
  
    for (size_t i = 0; i < ARRAYSIZE(candidates); i++) {
      const char *d = candidates[i];
      if (!d) continue;  // Empty env var
  
      // Make sure we don't surprise anyone who's expecting a '/'
      string dstr = d;
      if (dstr[dstr.size() - 1] != '/') {
        dstr += "/";
      }
      list->push_back(dstr);
  
      struct stat statbuf;
      if (!stat(d, &statbuf) && S_ISDIR(statbuf.st_mode)) {
        // We found a dir that exists - we're done.
        return;
      }
    }
  ```
  
  So it either defaults to `/tmp/` for all non windows system or, if either of the env variables `TEST_TMPDIR, TMPDIR, TMP" is defined, we choose that as the path. 
  
  And now here is my concern, the `user` naively does not pass the `log_dir` and somehow, they defined one of the `ENV` variables above (maybe for some other program). Then they decide to use this in production, so they spin up a new machine, and suddenly the `log_dir` defaults to something else because the env variables are not defined there. 
  
  Since we don't document this, and assuming we don't run on windows, should we just overwrite the flag in case it's `""` and default it to `/tmp/` ? This IMO would lead to less surprising behaviour

Reply:
We actually do run on Windows with WSL2 :)
Also, we support `epoll` (in addition to `uring`), and Roman is actively working on migration to MacOS and FreeBSD.

Reply:
More so, looking at an existing snapshot of a third party library and assuming that will remain the case in the future might be a bad idea. It's true that glog is a mature library, but still.. things can change, both in our supported environments (it's not hard to imagine an environment in which the temp directory lies elsewhere), or in glog.

I'm not sure that I understand why _not_ to call `GetLoggingDirectories()`. Can you please specify?

Reply:
`GetLoggingDirectories()` then :) 

P.s. The `envs` can still cause some surprising behavior. Maybe we should document this on the `--helpfull` command. What do you say?

Reply:
Yeah, it is a challenging task, because each of these paths can fail, and there's no way to determine the current path :(
We could try to phrase something like "logs are likely under this path: ...", but that doesn't sound too good I must say